stackslib/src/chainstate/burn/db/v2/test_helpers.rs

```diff
@@ -0,0 +1,1011 @@
+        &self,
+        block_id: u32,
+    ) -> Result<(u64, u64), crate::chainstate::stacks::index::Error> {
+        todo!()
+    }
+
+    fn get_external_trie_offset_length_by_bhh<T: crate::chainstate::stacks::index::MarfTrieId>(
+        &self,
+        bhh: &T,
+    ) -> Result<(u64, u64), crate::chainstate::stacks::index::Error> {
+        todo!()
+    }
+
+    fn get_external_blobs_length(&self) -> Result<u64, crate::chainstate::stacks::index::Error> {
+        todo!()
+    }
+
+    fn detect_partial_migration(&self) -> Result<bool, crate::chainstate::stacks::index::Error> {
+        todo!()
+    }
+
+    fn set_migrated(&self) -> Result<(), crate::chainstate::stacks::index::Error> {
+        todo!()
+    }
+
+    fn get_node_hash_bytes(
+        &self,
+        block_id: u32,
+        ptr: &crate::chainstate::stacks::index::node::TriePtr,
+    ) -> Result<stacks_common::types::chainstate::TrieHash, crate::chainstate::stacks::index::Error> {
+        todo!()
+    }
+
+    fn get_node_hash_bytes_by_bhh<T: crate::chainstate::stacks::index::MarfTrieId>(
+        &self,
+        bhh: &T,
+        ptr: &crate::chainstate::stacks::index::node::TriePtr,
+    ) -> Result<stacks_common::types::chainstate::TrieHash, crate::chainstate::stacks::index::Error> {
+        todo!()
+    }
+
+    fn tx_lock_bhh_for_extension<T: crate::chainstate::stacks::index::MarfTrieId>(
+        tx: &rusqlite::Connection,
+        bhh: &T,
+        unconfirmed: bool,
+    ) -> Result<bool, crate::chainstate::stacks::index::Error> {
+        todo!()
+    }
+
+    fn lock_bhh_for_extension<T: crate::chainstate::stacks::index::MarfTrieId>(
+        tx: &rusqlite::Transaction,
+        bhh: &T,
+        unconfirmed: bool,
+    ) -> Result<bool, crate::chainstate::stacks::index::Error> {
+        todo!()
+    }
+
+    fn count_blocks(&self) -> Result<u32, crate::chainstate::stacks::index::Error> {
+        todo!()
+    }
+
+    fn is_unconfirmed_block(&self, block_id: u32) -> Result<bool, crate::chainstate::stacks::index::Error> {
+        todo!()
+    }
+
+    fn drop_lock<T: crate::chainstate::stacks::index::MarfTrieId>(&self, bhh: &T) -> Result<(), crate::chainstate::stacks::index::Error> {
+        todo!()
+    }
+
+    fn drop_unconfirmed_trie<T: crate::chainstate::stacks::index::MarfTrieId>(&self, bhh: &T) -> Result<(), crate::chainstate::stacks::index::Error> {
+        todo!()
+    }
+
+    fn clear_lock_data(&self) -> Result<(), crate::chainstate::stacks::index::Error> {
+        todo!()
+    }
+
+    fn clear_tables(tx: &rusqlite::Transaction) -> Result<(), crate::chainstate::stacks::index::Error> {
+        todo!()
+    }
+
+    #[cfg(test)]
+    fn read_all_block_hashes_and_roots<T: crate::chainstate::stacks::index::MarfTrieId>(
+        &self,
+    ) -> Result<Vec<(stacks_common::types::chainstate::TrieHash, T)>, crate::chainstate::stacks::index::Error> {
+        todo!()
+    }
+}
+
+
+impl SortitionDb for InMemorySortitionDb {
```

Comment:
Same here -- this trait is way too big and encompasses several orthogonal concerns (some of which I don't think belong in the trait interface at all).

Reply:
Yeah, same answer here, I simply copy-pasted from impl blocks - any guidance here is welcomed :)